### PR TITLE
Widget: Text Widget

### LINKED ISSUE #1647: Text widget
It would be great to have a Text widget for creating consistent styled text without having to use CSS every time. This will make theming your application even easier.

Example usage,

```tsx
<Text size="h1">Heading</Text>
<Text style="success">Some lovely green text</Text>
```

You should be able to specify the size (i.e., header, subheader, normal body text, label text, etc) as well as a style like "success", "alert", etc.

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [x] For new widgets, an entry has been added to the `.dojorc`
* [x] For new widgets, `theme.variant()` is added to the root domnode
* [x] WidgetProperties are exported

**Description:**

`Text` widget used to provide a consistent approach to typography for a theme. The `Text` widget renders a `p` node by default that can be changed using the `as` property, for example:

```tsx
<Text as="h1">Heading Text</Text>
```

The font size is controlled by the `size` property (default to `m`) which in turn is defined by CSS vars in the theme variants. 

For longer text `truncated` can be used to add an ellipsis and finally `inverse` to change the text color to work on the inverse theme background.

Resolves #1647

### CODE REVIEW

File: src/text/index.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import theme from '../middleware/theme';
+
+import * as css from '../theme/default/text.m.css';
+
+export interface TextProperties {
+	size?: 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl';
+	inverse?: boolean;
+	truncated?: boolean;
+	as?: string;
+}
```
Comment (tomdye) [inline src/text/index.tsx:11]: Have you considered adding `weight` and `uppercase` ? I've found those useful on recent projects.

File: src/text/index.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { v, create } from '@dojo/framework/core/vdom';
+import theme from '../middleware/theme';
+
+import * as css from '../theme/default/text.m.css';
+
+export interface TextProperties {
+	size?: 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl';
+	inverse?: boolean;
+	truncated?: boolean;
+	as?: string;
```
Comment (tomdye) [inline src/text/index.tsx:10]: is it worth typing this to standard text nodes such as `span | div | h1-h6 | p` ?

Comment (agubler) [reply src/text/index.tsx:10]: I thought about it, but it really could be any node, unless that's something we want to restrict...?

File: src/text/index.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+const factory = create({ theme }).properties<TextProperties>();
+
+const Text = factory(function Text({ middleware: { theme }, properties, children }) {
+	const themedCss = theme.classes(css);
+	const { as = 'p', truncated = false, size = 'm', inverse = false } = properties();
```
Comment (tomdye) [inline src/text/index.tsx:17]: Where did the choice to default this to `p` come from? I'd have though that a `span` would be best as it's an inline text node then. Would make it useable more easily within all other widgets for consistent text styling, ie. within a `Button` / `Label` / `HelperText` etc.

Comment (agubler) [reply src/text/index.tsx:17]: I took inspiration from the Text component in Chakra that defaulted to `p`, open for discussion :)

File: src/text/index.tsx (hunk 1 above)
Comment (agubler) [reply src/text/index.tsx:11]: I think uppercase would be useful as a boolean that adds a text transform - Can you propose would you are thinking around weight?

Comment (tomdye) [reply src/text/index.tsx:11]: weight would follow the same as `size` ie. `s | m | l` etc and would apply `font-weight: 400 / 600 / 800` etc depending on the theme.

File: src/text/index.tsx (hunk 2 above)
Comment (tomdye) [reply src/text/index.tsx:10]: I guess it's better to leave as is then, but might have been nice for auto-completion

**Review by tomdye — APPROVED** [review]

**Pull request merged**